doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+
+- Marlowe contract transaction metadata (`MarloweTxMetadata`) - metadata attached to the on-chain transaction executing Marlowe Contract (`MarloweTx`) related to the `Contract`.
+
+
+### `Contract` extraction
+
+When initiator submits a `Contract` to the blockchain it builds up a transaction which outputs define role tokens distribution. Additionally transaction contains "continuation" output which is "guarded" by `Marlowe` interpreter script. Role tokens outputs are assigned to a set of addresses chosen by the `Tx` author. We can interpret them as invitations to the contract or permissions to be the contract party (this tokens shows up in the user wallets).
+
+Because continuation output from the initial `Tx` contains the full code ("modulo" `Merkleization`) of the `Contract` we are able to [extract it](https://github.com/input-output-hk/marlowe-cardano/blob/f452d98942a14d34c10f2963ec4b84147da5621c/marlowe/src/Language/Marlowe/Client/History.hs). In other words when someone is "invited" to the already initiated on-chain contract we are able to extract the Marlowe code from that transaction and present it to the possible participant for an audit, simulation etc. before he / she commits to it. We can do this at any stage of the contract progression because we can track the chain back to the initial transaction and perform the extraction.
+
+<!-- [ IMAGE -> Extraction of the contract -> Simulation / Audit ] -->
+
+### `MarloweTxMetadata`
+
+The submitter of the transaction have an ability to attach additional [`CBOR` encoded data to it](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md). This data are not accessible to the transaction validators so they can be arbitrary information not necessarily valid and related to the `Tx` content. To simplify the discussion for now and assume that there are no size limits and costs associated with `TxMetadata` and we use this layer to store all contract metadata.
+
+### `MarloweTxMetadata` verification
+
+We can imagine that we want to endorse a particular contract with a set of certified documents (agreements, audits etc.). We don't want to necessarily trust the user who creates the `MarloweTx` so we should assume that the contract code can be incosistent with `TxMetatdata` which provides the documents data. The simple protocol could require metadata to be signed together with the Marlowe contract so the connection between those pieces is clear.
+
+<!-- IMAGE -> Extraction of the contract -> Extraction of the metadata -->
+
+Because `Marlowe` abstracts over addresses using roles the full meaning of a particular instance of a contract is also dependent on role distribution. It seems that semantic meaning of a `Contract` on-chain is fully defined by the triple: `(Contract, RoleDistribution, MarloweValidator)`. Using a pseudo code we can provide this basic validation flow for `MarloweTxMetadata`:
```

Comment:
"full meaning of a particular instance of a contract" is a bit esoteric. I think it would be better to explain that we need to look at role distribution too in order to make sure that it was created correctly, because if, for example, if the creator of an escrow contract keeps the tokens for all parties it is not really an escrow contract.